access/grpc/grpc.go

```diff
@@ -645,3 +657,149 @@ func (c *BaseClient) GetExecutionResultForBlockID(ctx context.Context, blockID f
 		ServiceEvents:    serviceEvents,
 	}, nil
 }
+
+func (c *BaseClient) GetExecutionDataByBlockID(
+	ctx context.Context,
+	blockID flow.Identifier,
+	opts ...grpc.CallOption,
+) (*flow.ExecutionData, error) {
+
+	ed, err := c.executionDataClient.GetExecutionDataByBlockID(ctx, &executiondata.GetExecutionDataByBlockIDRequest{
+		BlockId: identifierToMessage(blockID),
+	}, opts...)
+	if err != nil {
+		return nil, newRPCError(err)
+	}
+
+	return messageToBlockExecutionData(ed.GetBlockExecutionData())
+
+}
+
+func (c *BaseClient) SubscribeExecutionData(
+	ctx context.Context,
+	startBlockID flow.Identifier,
+	startHeight uint64,
+	opts ...grpc.CallOption,
+) (<-chan flow.ExecutionDataStreamResponse, <-chan error, error) {
+
+	if startBlockID != flow.EmptyID && startHeight > 0 {
+		return nil, nil, fmt.Errorf("cannot specify both start block ID and start height")
+	}
+
+	req := executiondata.SubscribeExecutionDataRequest{}
+	if startBlockID != flow.EmptyID {
+		req.StartBlockId = startBlockID[:]
+	}
+	if startHeight > 0 {
+		req.StartBlockHeight = startHeight
+	}
+
+	stream, err := c.executionDataClient.SubscribeExecutionData(ctx, &req, opts...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sub := make(chan flow.ExecutionDataStreamResponse)
+	errChan := make(chan error)
+
+	go func() {
+		defer close(sub)
+		defer close(errChan)
+
+		for {
+			resp, err := stream.Recv()
+			if err == io.EOF {
+				return
+			}
+			if err != nil {
+				errChan <- fmt.Errorf("error receiving execution data: %w", err)
```

Comment:
since we have a context, it's possible the caller stopped monitoring the streams. It's probably a good idea to add a `select` for each of these channel operations